src/gas_price_increase.rs

```diff
@@ -31,15 +31,21 @@ fn new_gas_price_estimate(
 
 /// Adapt a stream of gas prices to only yield gas prices that respect the minimum gas price
 /// increase while filtering out other values, including those over the cap.
+/// Panics if gas price is negative or not finite.
 pub fn enforce_minimum_increase_and_cap(
     gas_price_cap: f64,
     stream: impl Stream<Item = f64>,
 ) -> impl Stream<Item = f64> {
-    let mut last_used_gas_price = 0.0;
+    let mut last_used_gas_price = None;
     stream.filter_map(move |gas_price| {
-        let gas_price = new_gas_price_estimate(last_used_gas_price, gas_price, gas_price_cap);
+        assert!(gas_price.is_finite() && gas_price >= 0.0);
+        let gas_price = if let Some(previous) = last_used_gas_price {
```

Comment:
I don't really understand the problem here. Wouldn't we just do another estimate if the gas price comes in at 0? What's the problem with that? Would it be easy to create a test case demonstrating the issue?

Maybe `new_gas_price_estimate` should operate take previous as an Option?

Reply:
In our e2e test in v2 the gas price is 0. The problem is if the real gas price is 0.0 without this fix we pass 0 as the previous and new gas price to new_gas_price_estimate which then filters out the new price because it has not increased. Edited the test.